### ISSUE TITLE
Page Whiteouts after opening Firefox Screenshots while holding the Escape button

### ISSUE THREAD

**Issue by emilghittasv** [mozilla-services/screenshots#2829]
[Affected versions]:
Firefox 54.0b6 (Build Id:20170508210007)
Firefox Screenshots 6.6.0

[Affected platforms]:
Windows 10 64 bit
MacOS 10.10.5 
Ubuntu 16.04 64bit

[Steps to reproduce]:
1. Open Firefox with a clean profile.
2. Enable Firefox Screenshots.
3. Access the https://reddit.com website.
4. Hold the keyboard’s Escape button.
5. Click the Firefox Screenshots button.

[Expected result]:
Firefox Screenshots does not open.

[Actual result]:
The entire page whiteouts.

[Additional notes]
Please note that you may have to repeat step 5 (while holding the Escape button) a couple of times to reproduce this issue.
Please observe the following video for more information: https://drive.google.com/open?id=0B94iuIVMr-TEWm13NmxjZWh5aFk

**Comment by johngruen** [comment]
Yeah, not sure exactly what's happening here, but it's gross. we should probably just disable screenshots if the user is depressing the esc key. I can repro this, but it seems to be maybe the result of a race condition b/c it doesn't happen every time, and seems to arbitrarily through push notification errors.

**Comment by johngruen** [comment]
Actually after doing this i'm getting constant error notifications every 5-10 seconds

**Comment by ianb** [comment]
I'm not getting quite the same errors, but I am seeing problems where the iframe isn't initialized in this case.  It's easy to imagine it could then produce regular errors.  I don't see any error that *causes* the iframe not to be initialized, the errors all come later.

My best first guess is that holding Escape causes something to try to tear down during initialization.

**Comment by ckprice** [comment]
Moving to stretch + may go away when we change button handler code.
